packages/angular_devkit/build_angular/src/builders/browser-esbuild/options.ts

```diff
@@ -204,6 +221,75 @@ export async function normalizeOptions(
+ * and if `main` is provided it will be used as the sole entry point. If `entryPoints` are provided, they will be used as the set of entry
+ * points.
+ *
+ * @param workspaceRoot Path to the root of the Angular workspace.
+ * @param main The `main` option pointing at the application entry point. While required per the schema file, it may be omitted by
+ *     programmatic usages of `browser-esbuild`.
+ * @param entryPoints Set of entry points to use if provided.
+ * @returns An object mapping entry point names to their file paths.
+ */
+function normalizeEntryPoints(
+  workspaceRoot: string,
+  main: string | undefined,
+  entryPoints: Set<string> = new Set(),
+): Record<string, string> {
+  if (main === '') {
+    throw new Error('`main` option cannot be an empty string.');
+  }
+
+  // `main` and `entryPoints` are mutually exclusive.
+  if (main && entryPoints.size > 0) {
+    throw new Error('Only one of `main` or `entryPoints` may be provided.');
+  }
+  if (!main && entryPoints.size === 0) {
+    // Schema should normally reject this case, but programmatic usages of the builder might make this mistake.
+    throw new Error('Either `main` or at least one `entryPoints` value must be provided.');
+  }
+
+  // Schema types force `main` to always be provided, but it may be omitted when the builder is invoked programmatically.
+  if (main) {
+    // Use `main` alone.
+    return { 'main': path.join(workspaceRoot, main) };
+  } else {
+    // Use `entryPoints` alone.
+    const entryPointPaths = new Map<string, string>();
+    for (const entryPoint of entryPoints) {
+      const parsedEntryPoint = path.parse(entryPoint);
+
+      // Use the input file path without an extension as the "name" of the entry point dictating its output location.
+      // Relative entry points are generated at the same relative path in the output directory.
+      // Absolute entry points are always generated with the same file name in the root of the output directory. This includes absolute
+      // paths pointing at files actually within the workspace root.
+      const entryPointName = path.isAbsolute(entryPoint)
+        ? parsedEntryPoint.name
+        : path.join(parsedEntryPoint.dir, parsedEntryPoint.name);
+
+      // Get the full file path to the entry point input.
+      const entryPointPath = path.isAbsolute(entryPoint)
+        ? entryPoint
+        : path.join(workspaceRoot, entryPoint);
+
+      // Check for conflicts with previous entry points.
+      const existingEntryPointPath = entryPointPaths.get(entryPointName);
+      if (existingEntryPointPath) {
+        throw new Error(
+          `\`${existingEntryPointPath}\` and \`${entryPointPath}\` both output to the same location \`${entryPointName}\`.` +
+            ' Rename or move one of the files to fix the conflict.',
+        );
+      }
+
+      entryPointPaths.set(entryPointName, entryPointPath);
+    }
+
+    return Object.fromEntries(entryPointPaths.entries());
```

Comment:
NIT: using a `Map` here is redundant. Consider using a key-value pair instead (same time complexity). 
This will also remove the need to convert the map to a key-value pair before returing.

```suggestion
    const entryPointPaths: Record<string, string> = {};
    for (const entryPoint of entryPoints) {
      const parsedEntryPoint = path.parse(entryPoint);

      // Use the input file path without an extension as the "name" of the entry point dictating its output location.
      // Relative entry points are generated at the same relative path in the output directory.
      // Absolute entry points are always generated with the same file name in the root of the output directory. This includes absolute
      // paths pointing at files actually within the workspace root.
      const entryPointName = path.isAbsolute(entryPoint)
        ? parsedEntryPoint.name
        : path.join(parsedEntryPoint.dir, parsedEntryPoint.name);

      // Get the full file path to the entry point input.
      const entryPointPath = path.isAbsolute(entryPoint)
        ? entryPoint
        : path.join(workspaceRoot, entryPoint);

      // Check for conflicts with previous entry points.
      const existingEntryPointPath = entryPointPaths[entryPointName];
      if (existingEntryPointPath) {
        throw new Error(
          `\`${existingEntryPointPath}\` and \`${entryPointPath}\` both output to the same location \`${entryPointName}\`.` +
            ' Rename or move one of the files to fix the conflict.',
        );
      }

      entryPointPaths[entryPointName] = entryPointPath;
    }

    return entryPointPaths;
```

Reply:
Good point, usually I prefer `Map`, but since we ultimately need to return an object, there's not much point in it here.